### PR TITLE
Api 5300 patient allow empty identifier

### PR DESCRIPTION
**Pull request by apeirats-va**

### CODE REVIEW

File: us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/Patient.java [hunk 1]
```diff
@@ -63,6 +63,11 @@
       message = "Only one multipleBirth field may be specified")
 })
 public class Patient implements Resource {
+
+  @JsonIgnore
```
Comment (bschofield-va) [inline us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/Patient.java:67]: Do you need `@JsonIgnore` on a static?

Comment (apeirats-va) [reply us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/Patient.java:67]: Guess not. Removing the `@JsonIgnore` did not affect the json output. Removing the `static final` did end up serializing this field to json, as expected.

File: us-core-r4/src/test/java/gov/va/api/health/r4/api/resources/PatientTest.java [hunk 2]
```diff
@@ -73,4 +78,19 @@ public void relatedGroups() {
...
+  @Test
+  public void validationFailsGivenNoIdentifiers() {
+    var patient = data.patient();
+    assertThat(violationsOf(patient)).isEmpty();
+    patient.identifier(List.of());
+    assertThat(violationsOf(patient)).isNotEmpty();
+    patient.identifier(null);
+    assertThat(violationsOf(patient)).isNotEmpty();
+  }
+
```
Comment (bschofield-va) [inline us-core-r4/src/test/java/gov/va/api/health/r4/api/resources/PatientTest.java:91]:
```
@Test void validationDoesNotFailWhenConfiguredWithMinSizeOfOne() {
  var save = Patient.IDENTIFIER_MIN_SIZE.get();
  try {
    Patient.IDENTIFIER_MIN_SIZE.set(0);
    // tests here
  } finally {
    Patient.IDENTIFIER_MIN_SIZE.set(save)
  }
}
```

Caveats to this approach... parallel tests can make this flakey, so we need to account for that. `@Execution(SAME_THREAD)` on the class should protect us.
https://junit.org/junit5/docs/snapshot/user-guide/index.html#writing-tests-parallel-execution-synchronization

File: us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/Patient.java [hunk 3]
```diff
@@ -143,6 +149,16 @@ private boolean isValidEthnicityExtension() {
         "http://hl7.org/fhir/us/core/StructureDefinition/us-core-ethnicity", 1);
   }
 
+  @JsonIgnore
+  @AssertTrue(message = "identifier does not meet size constraints")
```
Comment (jblefkove-va) [inline us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/Patient.java:153]: `identifier size is not valid` to match the other messages?

File: us-core-r4/src/test/java/gov/va/api/health/r4/api/resources/PatientTest.java [hunk 4]
```diff
@@ -73,4 +81,31 @@ public void relatedGroups() {
         .build()
         .verify();
   }
+
+  @Test
+  public void validationFailsGivenNoIdentifiers() {
```
Comment (jblefkove-va) [inline us-core-r4/src/test/java/gov/va/api/health/r4/api/resources/PatientTest.java:86]: unnecessary public

File: us-core-r4/src/test/java/gov/va/api/health/r4/api/resources/PatientTest.java [hunk 5]
```diff
@@ -73,4 +81,31 @@ public void relatedGroups() {
...
+    var save = Patient.IDENTIFIER_MIN_SIZE.get();
+    try {
+      Patient.IDENTIFIER_MIN_SIZE.set(0);
+      assertThat(violationsOf(patientNoIdentifier)).isEmpty();
+    } finally {
+      Patient.IDENTIFIER_MIN_SIZE.set(save);
+    }
+  }
+
+  private <T> Set<ConstraintViolation<T>> violationsOf(T object) {
```
Comment (jblefkove-va) [inline us-core-r4/src/test/java/gov/va/api/health/r4/api/resources/PatientTest.java:107]: should  be static

File: us-core-r4/src/test/java/gov/va/api/health/r4/api/resources/PatientTest.java [hunk 6]
```diff
@@ -73,4 +81,31 @@ public void relatedGroups() {
...
+    assertThat(violationsOf(patient)).isEmpty();
+    patient.identifier(List.of());
+    assertThat(violationsOf(patient)).isNotEmpty();
+    patient.identifier(null);
+    assertThat(violationsOf(patient)).isNotEmpty();
+  }
+
+  @Test
+  void validationPassesWhenConfiguredWithMinSizeOfZero() {
+    var patientNoIdentifier = data.patient().identifier(null);
```
Comment (bschofield-va) [inline us-core-r4/src/test/java/gov/va/api/health/r4/api/resources/PatientTest.java:97]: also this case: `patient.identifier(List.of());`

File: us-core-r4/src/test/java/gov/va/api/health/r4/api/resources/PatientTest.java (hunk 4 above)
Comment (apeirats-va) [reply us-core-r4/src/test/java/gov/va/api/health/r4/api/resources/PatientTest.java:86]: Removed unnecessary public scopes from the rest of the methods in this class

**Review by jblefkove-va — APPROVED** [review]

**Review by bschofield-va — APPROVED** [review]

**Pull request merged**